src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/TypeBuilderImpl.cs

```diff
@@ -0,0 +1,192 @@
+        [RequiresUnreferencedCode("P/Invoke marshalling may dynamically access members that could be trimmed.")]
+        protected override MethodBuilder DefinePInvokeMethodCore(string name, string dllName, string entryName, MethodAttributes attributes, CallingConventions callingConvention, Type? returnType, Type[]? returnTypeRequiredCustomModifiers, Type[]? returnTypeOptionalCustomModifiers, Type[]? parameterTypes, Type[][]? parameterTypeRequiredCustomModifiers, Type[][]? parameterTypeOptionalCustomModifiers, CallingConvention nativeCallConv, CharSet nativeCharSet) => throw new NotImplementedException();
+        protected override PropertyBuilder DefinePropertyCore(string name, PropertyAttributes attributes, CallingConventions callingConvention, Type returnType, Type[]? returnTypeRequiredCustomModifiers, Type[]? returnTypeOptionalCustomModifiers, Type[]? parameterTypes, Type[][]? parameterTypeRequiredCustomModifiers, Type[][]? parameterTypeOptionalCustomModifiers) => throw new NotImplementedException();
+        protected override ConstructorBuilder DefineTypeInitializerCore() => throw new NotImplementedException();
+        protected override FieldBuilder DefineUninitializedDataCore(string name, int size, FieldAttributes attributes) => throw new NotImplementedException();
+        protected override bool IsCreatedCore() => throw new NotImplementedException();
+        protected override void SetCustomAttributeCore(ConstructorInfo con, byte[] binaryAttribute) => throw new NotImplementedException();
+        protected override void SetCustomAttributeCore(CustomAttributeBuilder customBuilder) => throw new NotImplementedException();
+
+        protected override void SetParentCore([DynamicallyAccessedMembers((DynamicallyAccessedMemberTypes)(-1))] Type? parent)
+        {
+            if (parent != null)
+            {
+                if (parent.IsInterface)
+                    throw new ArgumentException(SR.Argument_CannotSetParentToInterface);
+
+                _typeParent = parent;
+            }
+            else
+            {
+                if ((_attributes & TypeAttributes.Interface) != TypeAttributes.Interface)
+                {
+#pragma warning disable IL2074 // Value stored in field does not satisfy 'DynamicallyAccessedMembersAttribute' requirements. The return value of the source method does not have matching annotations.
+                    _typeParent = _module.GetTypeFromCoreAssembly("System.Object");
+#pragma warning restore IL2074
+                }
+                else
+                {
+                    if ((_attributes & TypeAttributes.Abstract) == 0)
+                        throw new InvalidOperationException(SR.InvalidOperation_BadInterfaceNotAbstract);
+
+                    // there is no extends for interface class
```

Comment:
nit\inconsistentcy: some comments start with upper, some lower, and some have periods and some don't. I assume start with uppercase and end with period.